src/test/java/xogame/game_over/GameOverTest.java

```diff
@@ -34,118 +33,123 @@ public void check_content_of_the_board_by_rows() {
     }
 
     @Test
-    public void check_content_of_the_board_by_columns() {
-        Board board = new Board(new int[] { 2, 1, 0, 0, 2, 1, 1, 0, 2 });
-        assertEquals(Arrays.stream(new int[] { 2, 0, 1 }).boxed().collect(Collectors.toList()), board.getColumns().get(0));
-        assertEquals(Arrays.stream(new int[] { 1, 2, 0 }).boxed().collect(Collectors.toList()), board.getColumns().get(1));
-        assertEquals(Arrays.stream(new int[] { 0, 1, 2 }).boxed().collect(Collectors.toList()), board.getColumns().get(2));
+    public void check_binary_Operation() {
+        assertFalse(GameOver.isBitwiseOrAssignmentOperator(false, false));
+        assertTrue(GameOver.isBitwiseOrAssignmentOperator(false, true));
+        assertTrue(GameOver.isBitwiseOrAssignmentOperator(true, true));
     }
 
     @Test
-    public void the_first_row_is_complete_if_it_contains_all_ones() {
-        Board board = new Board(new int[] { 1, 1, 1, 2, 2, 0, 0, 0, 0 });
-        assertTrue(GameOver.isAnyRowComplete(board));
+    public void list_contains_one_empty_list_and_it_is_not_satisfy_to_the_condition() {
+        assertFalse(GameOver.isListSatisfyToTheCondition(Arrays.asList(emptyList), condition));
+    }  
+
+    @Test
+    public void list_contains_two_empty_lists_and_it_is_not_satisfy_to_the_condition() {
+        assertFalse(GameOver.isListSatisfyToTheCondition(Arrays.asList(emptyList, emptyList), condition));
     }
 
     @Test
-    public void the_first_row_is_complete_if_it_contains_all_twos() {
-        Board board = new Board(new int[] { 2, 2, 2, 1, 1, 0, 0, 0, 0 });
-        assertTrue(GameOver.isAnyRowComplete(board));
+    public void list_contains_one_not_empty_list_and_it_is_satisfy_to_the_condition() {
+        assertTrue(GameOver.isListSatisfyToTheCondition(Arrays.asList(Arrays.asList(1, 1, 1)), condition));
     }
 
     @Test
-    public void the_first_row_is_not_complete_if_it_contains_all_zeros() {
-        Board board = new Board(new int[] { 0, 0, 0, 2, 2, 0, 0, 0, 0 });
-        assertFalse(GameOver.isAnyRowComplete(board));
+    public void list_contains_two_not_empty_lists_and_it_is_satisfy_to_the_condition() {
+        assertTrue(GameOver.isListSatisfyToTheCondition(Arrays.asList(Arrays.asList(1, 1, 1), Arrays.asList(2, 2, 2)), condition));
     }
 
     @Test
-    public void the_second_row_is_complete_if_it_contains_all_ones() {
-        Board board = new Board(new int[] { 2, 2, 0, 1, 1, 1, 0, 0, 0 });
-        assertTrue(GameOver.isAnyRowComplete(board));
+    public void list_contains_one_empty_and_one_not_empty_lists_and_it_is_satisfy_to_the_condition() {
```

Comment:
I think it would be grammatically correct to say `list_contains_an_empty_and_a_non_empty_lists_and_it_satisfies_the_condition`.